aws/rust-runtime/aws-config/src/profile/timeout_config.rs

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Load timeout configuration properties from an AWS profile
+
+use crate::profile::Profile;
+use crate::provider_config::ProviderConfig;
+use aws_smithy_types::timeout::{TimeoutConfigBuilder, TimeoutConfigError};
+use aws_types::os_shim_internal::{Env, Fs};
+use std::time::Duration;
+
+const PROFILE_VAR_CONNECT_TIMEOUT: &str = "connect_timeout";
+const PROFILE_VAR_TLS_NEGOTIATION_TIMEOUT: &str = "tls_negotiation_timeout";
+const PROFILE_VAR_READ_TIMEOUT: &str = "read_timeout";
+const PROFILE_VAR_API_CALL_ATTEMPT_TIMEOUT: &str = "api_call_attempt_timeout";
+const PROFILE_VAR_API_CALL_TIMEOUT: &str = "api_call_timeout";
+
+/// Load timeout configuration properties from a profile file
+///
+/// This provider will attempt to load AWS shared configuration, then read timeout configuration properties
+/// from the active profile.
+///
+/// # Examples
+///
+/// **Sets the `connect_timeout` to 2 seconds
+/// ```ini
+/// [default]
+/// connect_timeout = 2
+/// ```
+///
+/// **Sets the `connect_timeout` to 0.5 seconds _if and only if_ the `other` profile is selected.
+///
+/// ```ini
+/// [profile other]
+/// connect_timeout = 0.5
+/// ```
+///
+/// This provider is part of the [default timeout_config provider chain](crate::default_provider::timeout_config).
+#[derive(Debug, Default)]
+pub struct ProfileFileTimeoutConfigProvider {
+    fs: Fs,
+    env: Env,
+    profile_override: Option<String>,
+}
+
+/// Builder for [ProfileFileTimeoutConfigProvider]
+#[derive(Default)]
+pub struct Builder {
+    config: Option<ProviderConfig>,
+    profile_override: Option<String>,
+}
+
+impl Builder {
+    /// Override the configuration for this provider
+    pub fn configure(mut self, config: &ProviderConfig) -> Self {
+        self.config = Some(config.clone());
+        self
+    }
+
+    /// Override the profile name used by the [ProfileFileTimeoutConfigProvider]
+    pub fn profile_name(mut self, profile_name: impl Into<String>) -> Self {
+        self.profile_override = Some(profile_name.into());
+        self
+    }
+
+    /// Build a [ProfileFileTimeoutConfigProvider] from this builder
```

Comment:
In general, name links end up looking better in the docs if you wrap them with backticks as well. For example:
```
[`ProfileFileTimeoutConfigProvider`]
```